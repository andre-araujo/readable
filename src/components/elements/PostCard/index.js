import React, { Component, Fragment } from 'react';
import { Link } from 'react-router-dom';

function PostCard ({
  id,
  timestamp,
  title,
  body,
  author,
  category,
  voteScore
}) {
    return (
      <section>
        <header>
          <h1>{title}</h1>
          <p>By: {author}</p>
        </header>
        <div>
          <p>{body}</p>
        </div>
        <footer>
          <time>{timestamp}</time>
          <p>{voteScore}</p>
        </footer>
      </section>
    );
}

export default PostCard;
