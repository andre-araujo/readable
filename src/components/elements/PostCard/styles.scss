.container {
  padding: 20px;
  background: #fafafa;
  font-family: AvantGardeLight
}

.footer {
  padding-top: 10px;
  margin-top: 10px;
  border-top: 1px solid #ccc;
  display: flex;
  justify-content: space-between;
}

.time {
  font-size: 14px;
  color: #666;
}

.title {
  font-family: AvantGardeDemi;
  text-transform: uppercase;
  margin-bottom: 5px;
  font-size: 18px;
}

.author {
  font-family: AvantGardeMedium;
  font-size: 12px;
  margin-bottom: 15px;
}
