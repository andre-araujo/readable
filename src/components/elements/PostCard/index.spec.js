import React from 'react';
import { shallow } from 'enzyme';

import PostCard from '.';

describe('PostCard', () => {
  it('should match snapshot', () => {
    expect(shallow(<PostCard />)).toMatchSnapshot();
  });
});
