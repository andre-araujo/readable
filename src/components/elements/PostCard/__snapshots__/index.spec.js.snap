// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard should match snapshot 1`] = `
<section>
  <header>
    <h1 />
    <p>
      By: 
    </p>
  </header>
  <div>
    <p />
  </div>
  <footer>
    <time />
    <p />
  </footer>
</section>
`;
